src/protocol/model.rs

```diff
@@ -194,9 +194,19 @@ pub mod auth {
     #[derive(Debug, Clone, Serialize, Deserialize)]
     pub struct AuthResponseContent {
         pub challenge: String,
-        pub granted_permissions: Vec<String>,
-        pub session_token: String,
         pub subkey_proof: Option<SubkeyProof>,
+        pub status: AuthResponseStatus,
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize)]
+    pub enum AuthResponseStatus {
```

Comment:
You will have to expose this with uniffi to allow the app to construct this enum. You can copy from other enums, usually it's just `#[derive(uniffi::Enum)]`